ci-operator/step-registry/cucushift/installer/check/azure/private/lb/cucushift-installer-check-azure-private-lb-commands.sh

```diff
@@ -0,0 +1,125 @@
+    az cloud set --name AzureUSGovernment
+elif [[ "${CLUSTER_TYPE}" == "azurestack" ]]; then
+    if [ ! -f "${CLUSTER_PROFILE_DIR}/cloud_name" ]; then
+        echo "Unable to get specific ASH cloud name!"
+        exit 1
+    fi
+    cloud_name=$(< "${CLUSTER_PROFILE_DIR}/cloud_name")
+
+    AZURESTACK_ENDPOINT=$(cat "${SHARED_DIR}"/AZURESTACK_ENDPOINT)
+    SUFFIX_ENDPOINT=$(cat "${SHARED_DIR}"/SUFFIX_ENDPOINT)
+
+    if [[ -f "${CLUSTER_PROFILE_DIR}/ca.pem" ]]; then
+        cp "${CLUSTER_PROFILE_DIR}/ca.pem" /tmp/ca.pem
+        cat /usr/lib64/az/lib/python*/site-packages/certifi/cacert.pem >> /tmp/ca.pem
+        export REQUESTS_CA_BUNDLE=/tmp/ca.pem
+    fi
+    az cloud register \
+        -n ${cloud_name} \
+        --endpoint-resource-manager "${AZURESTACK_ENDPOINT}" \
+        --suffix-storage-endpoint "${SUFFIX_ENDPOINT}"
+    az cloud set --name ${cloud_name}
+    az cloud update --profile 2019-03-01-hybrid
+else
+    az cloud set --name AzureCloud
+fi
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+INFRA_ID=$(jq -r .infraID ${SHARED_DIR}/metadata.json)
+RESOURCE_GROUP=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.resourceGroupName')
+if [[ -z "${RESOURCE_GROUP}" ]]; then
+    RESOURCE_GROUP="${INFRA_ID}-rg"
+fi
+
+publish=$(yq-go r "${INSTALL_CONFIG}" "publish")
+outbound_type=$(yq-go r "${INSTALL_CONFIG}" "platform.azure.outboundType")
+echo -e "--- cluster config --- \npublish: ${publish}\noutbound_type: ${outbound_type}"
+
+if [[ "${publish}" != "Internal" ]]; then
+    echo "This step is applicable on azure private cluster!"
+    exit 1
+fi
+
+critical_check_result=0
+no_critical_check_result=0
+
+#lb check
+lb_list=$(az network lb list -g ${RESOURCE_GROUP} -ojson |jq -r '.[].name')
+echo "INFO: lb list is ${lb_list}"
+lb_internal="${INFRA_ID}-internal"
+if [[ "${lb_list}" =~ ${lb_internal} ]]; then
+    lb_public=$(echo ${lb_list/"${lb_internal}"} | tr -d '\n')
+else
+    echo "ERROR: unable to find internal lb ${lb_internal}!"
+    exit 1
+fi
+
+if [[ "${outbound_type}" == "UserDefinedRouting" ]]; then
+    echo "Cluster outBound type is UserDefinedRouting, this is fully private cluster..."
+    if (( ocp_minor_version >= 11 )); then
+        echo "Check that no public load balancer created on fully private cluster..."
+        [[ -n "${lb_public}" ]] && echo "WARN: Found public load balancer ${lb_public} on fully private cluster!" && no_critical_check_result=1
+    fi
+    # check lb frontended IP config
+    for lb in ${lb_list}; do
+        echo "Check public ip address of Loadbalance ${lb}, should not be created..."
+        ret=0
+        az network lb show -g ${RESOURCE_GROUP} -n ${lb} | jq -r '.frontendIPConfigurations[].publicIPAddress.id' && ret=1
+        [[ ${ret} -eq 0 ]] && echo "ERROR: Found public ip address for load balancer: ${lb}" && critical_check_result=1
```

Comment:
```suggestion
        lb_public_ip=""
        lb_public_ip=$(az network lb show -g ${RESOURCE_GROUP} -n ${lb} | jq -r '.frontendIPConfigurations[].publicIPAddress.id')
        [[ -n "${lb_public_ip}" ]] && echo "ERROR: Found public ip address for load balancer: ${lb}" && critical_check_result=1
```

How about this for easy reading.

Reply:
if az return `null`, we can append `[[ "${lb_public_ip}" != "null" ]]` to `[[ -n "${lb_public_ip}" ]]`